javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -20,15 +26,46 @@ public final class OpenTelemetryInstaller {
    * @return the {@link AutoConfiguredOpenTelemetrySdk}
    */
   public static AutoConfiguredOpenTelemetrySdk installOpenTelemetrySdk(
-      ClassLoader extensionClassLoader) {
+      ClassLoader extensionClassLoader, EarlyInitAgentConfig earlyConfig) {
 
     AutoConfiguredOpenTelemetrySdk autoConfiguredSdk =
         AutoConfiguredOpenTelemetrySdk.builder()
             .setResultAsGlobal()
             .setServiceClassLoader(extensionClassLoader)
             .build();
+    ConfigProvider configProvider = AutoConfigureUtil.getConfigProvider(autoConfiguredSdk);
     OpenTelemetrySdk sdk = autoConfiguredSdk.getOpenTelemetrySdk();
 
+    setForceFlush(sdk);
+
+    if (configProvider != null) {
+      // We create a new instance of AutoConfiguredOpenTelemetrySdk, which has a ConfigProperties
+      // instance that can be used to read properties from the configuration file.
+      // This allows most instrumentations to be unaware of which configuration style is used.
+      return SdkAutoconfigureAccess.create(
+          sdk,
+          // the Resource from the original instance is also the default Resource,
+          // so this is functionally equivalent
+          Resource.getDefault(),
```

Comment:
This is not a good explanation. My understanding is that it is only functionally equivalent because when using declarative config we can't extract resource through the `getResource` method. Although it is a package private method and not a public api we could argue is a shortcoming in the opentelemetry-java. Once that is addressed this comment will be wrong. Consider extracting the resource form the original `autoConfiguredSdk` so that once `getResource` starts working we'd be handling the resource correctly here too.